cmd/collector/app/processor/processor.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package processor
+
+import (
+	"io"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+var (
+	_ Batch = (*SpansV1)(nil)
+	_ Batch = (*SpansV2)(nil)
+)
+
+// Batch is a batch of spans passed to the processor.
+type Batch interface {
+	// GetSpans delegates to the appropriate function based on the data model version.
+	GetSpans(v1 func(spans []*model.Span), v2 func(traces ptrace.Traces))
+
+	GetSpanFormat() SpanFormat
+	GetInboundTransport() InboundTransport
+	GetTenant() string
+}
+
+// SpanProcessor handles spans
```

Comment:
can we make this more detailed?

Reply:
There isn't much more meaning to it at the interface level - the actual semantics are in the collector code. BTW this is all v1 code, I don't want to invest a lot of time in it, but I do want to fix it to be as performance with v2 model as with v1 (per the ticket).